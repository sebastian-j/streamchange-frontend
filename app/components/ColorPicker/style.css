.color-picker-wrapper {
    display: flex;
    flex-direction: row;
    margin: 3px;
}

.color-picker-button {
    border: 1px gray solid;
    border-radius: 50%;
    height: 30px;
    width: 30px;
    outline: none;
}

.color-picker-label {
    margin: 5px 10px 0 0;
}

.inner-picker {
    position: absolute;
    z-index: 100;
}

.active-picker-background {
    background-color: rgba(0, 0, 0, 0.2);
    border: none;
    position: fixed;
    left: 0;
    top: 0;
    height: 100%;
    width: 100%;
    outline: none;
}