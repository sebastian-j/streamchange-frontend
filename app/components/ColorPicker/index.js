import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { SketchPicker } from 'react-color';
import './style.css';

const ColorPicker = (props) => {
  const [color, setColor] = useState(props.color);
  const [colorString, setColorString] = useState(props.color);
  const [pickerVisible, setPickerVisible] = useState(false);

  useEffect(() => {
    setColorString(props.color);
  }, [color]);

  const handleColorChange = (colorObject) => {
    const rgba = colorObject.rgb;
    const hex = `rgba(${rgba.r}, ${rgba.g}, ${rgba.b}, ${rgba.a})`;
    setColor(rgba);
    setColorString(hex);
    props.handleChange(props.name, hex);
  };

  const onTogglePicker = () => {
    setPickerVisible((prevState) => !prevState);
  };
  return (
    <div className="color-picker-wrapper">
      <div className="color-picker-label">{props.label}</div>
      <button
        aria-label="toggle picker"
        className="color-picker-button"
        style={{ backgroundColor: colorString }}
        onClick={onTogglePicker}
        type="button"
      />
      {pickerVisible && (
        <div>
          <button
            aria-label="color picker"
            className="active-picker-background"
            onClick={onTogglePicker}
            type="button"
          />
          <div style={{ position: 'absolute' }}>
            <SketchPicker
              className="inner-picker"
              color={color}
              onChangeComplete={handleColorChange}
            />
          </div>
        </div>
      )}
    </div>
  );
};

ColorPicker.propTypes = {
  color: PropTypes.string,
  name: PropTypes.string,
  label: PropTypes.string,
  handleChange: PropTypes.func,
};

export default ColorPicker;
