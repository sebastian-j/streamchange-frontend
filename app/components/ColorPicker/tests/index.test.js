import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import ColorPicker from '../index';

const renderer = new ShallowRenderer();

describe('<ColorPicker />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<ColorPicker />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
