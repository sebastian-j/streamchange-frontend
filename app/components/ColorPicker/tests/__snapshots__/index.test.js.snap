// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorPicker /> should render and match the snapshot 1`] = `
<div
  className="color-picker-wrapper"
>
  <div
    className="color-picker-label"
  />
  <button
    className="color-picker-button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
    type="button"
  />
</div>
`;
